docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1190 @@
+   }
+   else{
+      printf("User % not found.\n", nm);
+   } 
+}
+
+ACTION hello::deleterow(name nm) {
+   user_data_table users(get_self(), get_self().value);
+
+   auto itr = users.find(nm.value);
+   if ( itr != users.end() ) {
+      users.erase(itr);
+      printf("User % erased.\n", nm);
+   }
+   else{
+      printf("User % not found.\n", nm);
+   }
+}
+```
+
+### Multi-index: Test
+
+Build and deploy again the smart contract, send the `createrow` action a couple of times and observe the results:
+
+```shell
+dune --send-action hello createrow '[ama]' hello@active
+```
+
+```txt
+#         hello <= hello::createrow             {"nm":"ama"}
+>> User added as non-admin.
+```
+
+```shell
+dune --send-action hello createrow '[ama]' hello@active
+```
+
+```txt
+#         hello <= hello::createrow             {"nm":"ama"}
+>> User already exists.
+```
+
+Note how the first action created the a non-admin user `ama` and the second one did not because the user already existed.
+
+Read the `ama` user data:
+
+```shell
+dune --send-action hello readrow '[ama]' hello@active
+```
+
+```txt
+#         hello <= hello::readrow               {"nm":"ama"}
+>> User non-admin ama found.
+```
+
+Make the user `ama` admin:
+
+```shell
+dune --send-action hello updaterow '[ama, 1]' hello@active
+```
+
+```txt
+#         hello <= hello::updaterow             {"nm":"ama","is_admin":1}
+>> User ama is_admin was set to true.
+```
+
+Delete the user `ama`:
+
+```shell
+dune --send-action hello deleterow '[ama]' hello@active
+```
+
+```txt
+#         hello <= hello::deleterow             {"nm":"ama"}
+>> User erased.
+```
+
+## Singleton
+
+A singleton is a special multi-index table that is designed to store a single row of data for each instance of the singleton type. Singletons are often used to store global state variables or configuration parameters in a contract.
+
+It is important to remember the explanation of code and scope. When you instantiate a singleton, you can keep the code parameter fixed, and vary the scope parameter. This way you save one item per scope, and thus for example, you can store per-account configs.
+
+Here's an example of a singleton definition:
+
+```cpp
+TABLE statsdata {
+    int count;
+};
+using stats_singleton = eosio::singleton<"stats"_n, statsdata>;
+```
+
+The code above defines a singleton type `stats_singleton`. This singleton stores statistical data defined by `statsdata` structure. The structure contains the `count` data member which can hold an arbitrary integer value.
+
+Developers can use the `stats_singleton` template type, to instantiate a reference of the singleton table and to perform various operations, such as:
+
+- read the singleton data,
+- modify existing singleton data,
+- delete existing singleton data.
+
+### Singleton: Code and Scope
+
+The code and scope have the same meaning as for the [multi-index table](#multi-index-code-and-scope).
+
+#### Get Singleton Data
```

Comment:
Same thing with these example sections. This teaching is duplicated below with the hello contract updates